5.0 Draft/IDD/IDDs Orchestration System/eu.arrowhead.orchestration-pull-http-json.yml

```diff
@@ -0,0 +1,303 @@
+
+    In the context of this document, we reffer to the Orchestration consumer as _requesting-system_ where _system-id_ is the
+    unique identifier that represents that system. The _requested-service_ represents the service information that the _requesting-system_
+    requires orchestration. In case the _requested_service_ is specified, the requires service definition must be included in the form, 
+    in addition, at least one interface must be specified. The orchestration system will attempt to match the _requested-service_ information
+    with services present on the orchestration rules or SoS configurations. 
+
+version: 5.0.0-rc.1
+
+# Additional information present in the IDD template
+x-transfer-protocol: [HTTP, HTTPS]
+x-encryption-method: to-add
+x-encoding: JSON
+x-compression: to-add
+x-semantics: to-add
+x-Ontology: to-add
+
+paths:
+  /orchestration/echo:
+    get:  
+      summary: Endpoint to check the core system availability. 
+      description: Returns an OK to indicate that the core system is available. 
+      responses:
+        '200':
+          description: OK
+
+
+  /orchestration/orchestration-pull:
+    post:
+      summary: Submit a service request and get orchestration response. If no service definition is specified, the orchestration
+        will return the suggested services and their appropriate providers for the consuming system to fulfil its tasks, depending
+        on a existing SoS configuration or Orchestration rules. 
+
+        Information of the requesting system is mandatory. Requested service is optional. If a requested service object is specified,
+        the service definition and at least one interface must be specified. 
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ServiceRequestForm'
+      responses:
+        '200':
+          description: Successful orchestration response.
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/OrchestrationResponse'
+
+  /orchestration/orchestration-pull/{id}:
+    get:
+      summary: Get Orchestration by system id
+      parameters:
+        - name: id
+          in: path
+          required: true
+          description: ID of requesting system
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/OrchestrationResponse'
+        '404':
+          description: System id not found
+          content:
+            application/json:
+              example:
+                error: Orchestration failed, system id not found. 
+
+
+components:
+  schemas:
+    Address:
```

Comment:
I remember @PerOlofsson-Sinetiq and I concluding that we don't want the service orchestration service to deal with address information at all. You should be given names you can look up in the service registry after having consulted the service orchestration service. All other details the consumer has to look up by itself.

Reply:
Indeed! 

The overall idea is that the application systems should be served by the core systems, meaning that the application systems will have the initiative and request information from the core systems.
 
This might be a contradiction to the AITIA reference implementation? My understanding is that they rather have the core systems controlling the behaviour on the applications than the other way around.